bootstrap/src/basis/u64.c

```diff
@@ -0,0 +1,20 @@
+#include <stddef.h>
+#include <caml/mlvalues.h>
+
+CAMLprim value
+hemlock_u64_bit_pop(value a_x) {
+  uint64_t x = Int64_val(a_x);
+  return Val_long(__builtin_popcountl(x));
+}
+
+CAMLprim value
+hemlock_u64_bit_clz(value a_x) {
+  uint64_t x = Int64_val(a_x);
+  return Val_long((x == 0) ? 64 : __builtin_clzl(x));
+}
+
+CAMLprim value
+hemlock_u64_bit_ctz(value a_x) {
+  uint64_t x = Int64_val(a_x);
+  return Val_long((x == 0) ? 64 : __builtin_ctzl(x));
```

Comment:
Why does the check for zero need to be here? Guessing this speeds up one of our existing use cases?

Reply:
The result of `__builtin_ctzl()` is undefined for the `(x == 0)` case.  I didn't initially notice it in the documentation (it _is_ documented), but the unit tests caught the bug.